google-ads-examples/src/main/java/com/google/ads/googleads/examples/planning/ForecastReach.java

```diff
@@ -0,0 +1,357 @@
+      for (ReachPlanAgeRange ageRange : product.getPlannableTargeting().getAgeRangesList()) {
+        System.out.printf("\t- %s%n", ageRange);
+      }
+      System.out.println("Genders:");
+      for (GenderInfo gender : product.getPlannableTargeting().getGendersList()) {
+        System.out.printf("\t- %s%n", gender.getType());
+      }
+      System.out.println("Devices:");
+      for (DeviceInfo device : product.getPlannableTargeting().getDevicesList()) {
+        System.out.printf("\t- %s%n", device.getType());
+      }
+    }
+  }
+
+  /**
+   * Creates a base request to generate a reach forecast.
+   *
+   * @param customerId the customer ID for the reach forecast.
+   * @param productMix the product mix for the reach forecast.
+   * @param locationId location ID to plan for. To find a valid location ID, either see
+   *     https://developers.google.com/adwords/api/docs/appendix/geotargeting or call
+   *     ReachPlanServiceClient.ListPlannableLocations.
+   * @param currencyCode three-character ISO 4217 currency code.
+   * @return populated GenerateReachForecastRequest object.
+   */
+  private GenerateReachForecastRequest buildReachRequest(
+      Long customerId, List<PlannedProduct> productMix, String locationId, String currencyCode) {
+    CampaignDuration duration =
+        CampaignDuration.newBuilder().setDurationInDays(Int32Value.of(28)).build();
+
+    List<GenderInfo> genders =
+        Arrays.asList(
+            GenderInfo.newBuilder().setType(GenderType.FEMALE).build(),
+            GenderInfo.newBuilder().setType(GenderType.MALE).build());
+
+    List<DeviceInfo> devices =
+        Arrays.asList(
+            DeviceInfo.newBuilder().setType(Device.DESKTOP).build(),
+            DeviceInfo.newBuilder().setType(Device.MOBILE).build(),
+            DeviceInfo.newBuilder().setType(Device.TABLET).build());
+
+    Targeting targeting =
+        Targeting.newBuilder()
+            .setPlannableLocationId(StringValue.of(locationId))
+            .setAgeRange(ReachPlanAgeRange.AGE_RANGE_18_65_UP)
+            .addAllGenders(genders)
+            .addAllDevices(devices)
+            .build();
+
+    // See the docs for defaults and valid ranges:
+    // https://developers.google.com/google-ads/api/reference/rpc/Google.Ads.GoogleAds.V4.services#Google.Ads.GoogleAds.V4.services.GenerateReachForecastRequest
+    return GenerateReachForecastRequest.newBuilder()
+        .setCustomerId(Long.toString(customerId))
+        .setCurrencyCode(StringValue.of(currencyCode))
+        .setCampaignDuration(duration)
+        .setTargeting(targeting)
+        .setMinEffectiveFrequency(Int32Value.of(1))
+        .addAllPlannedProducts(productMix)
+        .build();
+  }
+
+  /**
+   * Pulls and prints the reach curve for the given request.
+   *
+   * @param reachPlanServiceClient instance of Reach Plan Service client.
+   * @param request an already-populated reach curve request.
+   */
+  private void pullReachCurve(
```

Comment:
How about "get" or "retrieve" instead of "pull"?

Reply:
Defer to @AnashOommen who wrote the golden.

Reply:
Defer to @AnashOommen who wrote the golden.

Reply:
GetReachCurve is fine.